internal/provider/resource_gitlab_personal_access_tokens.go

```diff
@@ -0,0 +1,242 @@
+				ValidateDiagFunc: isISO6801Date,
+			},
+			"token": {
+				Description: "The personal access token. This is only populated when creating a new personal access token. This attribute is not available for imported resources.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+})
+
+func resourceGitlabPersonalAccessTokenCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	userId := d.Get("user_id").(int)
+
+	currentUserAdmin, err := isCurrentUserAdmin(client)
+	if err != nil {
+		return diag.Errorf("[ERROR] cannot query the user API for current user: %v", err)
+	}
+
+	if !currentUserAdmin {
+		return diag.Errorf("[ERROR] Current user needs to be admin when creating a personal access token: %v", err)
+	}
+
+	options := &gitlab.CreatePersonalAccessTokenOptions{
+		Name:   gitlab.String(d.Get("name").(string)),
+		Scopes: stringSetToStringSlice(d.Get("scopes").(*schema.Set)),
+	}
+
+	log.Printf("[DEBUG] create gitlab PersonalAccessToken %s (scopes: %s) for user ID %d", *options.Name, options.Scopes, userId)
+
+	if v, ok := d.GetOk("expires_at"); ok {
+		parsedExpiresAt, err := time.Parse("2006-01-02", v.(string))
+		if err != nil {
+			return diag.Errorf("Invalid expires_at date: %v", err)
+		}
+
+		parsedExpiresAtISOTime := gitlab.ISOTime(parsedExpiresAt)
+		options.ExpiresAt = &parsedExpiresAtISOTime
+	}
+
+	personalAccessToken, _, err := client.Users.CreatePersonalAccessToken(userId, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tokenId := strconv.Itoa(personalAccessToken.ID)
+	userIdString := strconv.Itoa(userId)
+
+	d.SetId(buildTwoPartID(&userIdString, &tokenId))
+	// NOTE: the token can only be read once after creating it
+	d.Set("token", personalAccessToken.Token)
+
+	return resourceGitlabPersonalAccessTokenRead(ctx, d, meta)
+}
+
+func resourceGitlabPersonalAccessTokenRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	userId, tokenId, err := parseTwoPartID(d.Id())
+	if err != nil {
+		return diag.Errorf("Error parsing ID: %s", d.Id())
+	}
+
+	userIdInt, err := strconv.Atoi(userId)
+	if err != nil {
+		return diag.Errorf("Error converting user ID to string: %v", userId)
+	}
+
+	client := meta.(*gitlab.Client)
+
+	personalAccessTokenId, err := strconv.Atoi(tokenId)
+	if err != nil {
+		return diag.Errorf("%s cannot be converted to int", tokenId)
+	}
+
+	log.Printf("[DEBUG] read gitlab PersonalAccessToken %d, user ID %s", personalAccessTokenId, userId)
+
+	personalAccessToken, err := resourceGitlabPersonalAccessTokenFind(ctx, client, userIdInt, personalAccessTokenId)
+	if errors.Is(err, errResourceGitlabPersonalAccessTokenNotFound) {
+		log.Printf("[DEBUG] failed to read gitlab PersonalAccessToken %d, user ID %s", personalAccessTokenId, userId)
+		d.SetId("")
+	}
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
NIT: not a big fan of the readability here. I suggest to avoid fallthroughs like that and early exit:

```suggestion
	if errors.Is(err, errResourceGitlabPersonalAccessTokenNotFound) {
		log.Printf("[DEBUG] failed to read gitlab PersonalAccessToken %d, user ID %s", personalAccessTokenId, userId)
		d.SetId("")
		return nil
	}

	if err != nil {
		return diag.FromErr(err)
	}
```

Reply:
... it's even an issue, right?!

Reply:
Should I also remove `[DEBUG]` like I did with `[ERROR]` or do debug statements print differently?

Reply:
AFAIK: debug statements are different ...

Reply:
Sounds good this is good to go then